misaligned.cpp

```diff
@@ -1,21 +1,44 @@
 #include <assert.h>
 #include <iostream>
+#include <string>
+#include <sstream>
+#include <vector>
+using  std::stringstream, std::ios_base, std::string, std::cout, std::vector;
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
+const char* majorColor[] = { "White", "Red", "Black", "Yellow", "Violet" };
+const char* minorColor[] = { "Blue", "Orange", "Green", "Brown", "Slate" };
+stringstream printColorMap() {
+    stringstream message("\n", ios_base::app | ios_base::out);
     int i = 0, j = 0;
     for (i = 0; i < 5; i++) {
         for (j = 0; j < 5; j++) {
             std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
         }
     }
-    return i * j;
+    cout << message.str();
+    return message;
 }
 
 int main() {
-    int result = printColorMap();
-    assert(result == 25);
+    std::string outtext;
+    stringstream result(printColorMap().str());
+    string line;
+    vector<string> stringList;
+    // Loop until the end of the string
+    while (getline(result, line)) {
+        if (!line.empty()) {
+            stringList.push_back(line);
+        }
+    }
+    for (int i = 0; i < 5; i++) {
+        for (int j = 0; j < 5; j++) {
+            int code = (i * 5+ j);
+            string stringLine = stringList.at(code);
+            assert(stringLine.find(std::to_string(code+1)) != string::npos);

```

Comment:
Careful with the assert... this will pass even if `11` is present instead of `1`.
Think of the user-need while thinking of the assert. Will the user be satisfied if the string form of `code+1` is present "somewhere" in the output? (that's what the assert says here)